api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +623,50 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  '/event-types/{name}/cursors':
+      get:
+        tags:
+          - unmanaged-api
+          - monitoring
+          - management-api
+        security:
+          - oauth2: ['nakadi.event_stream.read']
+        description: |
+          Transforms a list of Cursors with `shift` into a list without `shift`s. This is useful when there is the need
+          for randomly access events in the stream.
+
+        parameters:
+          - name: name
+            in: path
+            description: EventType name
+            type: string
+            required: true
+          - name: X-Flow-Id
+            in: header
+            description: |
+              The flow id of the request, which is written into the logs and passed to called services. Helpful
+              for operational troubleshooting and log analysis.
+            type: string
+          - name: X-nakadi-cursors
+            in: header
+            description: |
+              Cursors indicating the positions to be calculated. If a cursors contains `shift` property, this property
+              will be used to calculate the actual offset for the given position and in the response the cursor will
+              no longer contain the `shift`.
+            required: true
+            type: array
+            items:
+              type: string
+              format: '#/definitions/ShiftedCursor'
+        responses:
+          '200':
+            description: OK
+            schema:
+              type: array
+              description: An array of `Cursor`s with shift applied to the offset.
```

Comment:
I don't think we need to promise any order in the API. Since the returned object contains all the information about the query (I mean, `ShiftedCursor` has `partition` and `offset`) so that the requester don't need to rely on order.

I added 422 to the response.